### DISCUSSION TITLE
Different App Icon/ Splash / plist / manifest files for different build configurations. Possible???

### DISCUSSION THREAD

**Discussion by LeoJHarris** [dotnet/maui#24622] [Q&A] [answered]
Hi All,

Can somebody please provide assistance for enabling different AppIcon/Splash screen for different build configurations:

@jfversluis had a blog all about this including a sample repo:
https://github.com/jfversluis/DifferentAppIconsSample

https://web.archive.org/web/20230327010313/https://blog.verslu.is/xamarin/different-app-icons-for-different-configurations-in-xamarin/ (old blog now returns 404)

Inline with the old way of doing this in XF I was doing the following:

```csproj
<ItemGroup ...>
    <None Include="InfoCustomOne.plist">
          <LogicalName>Info.plist</LogicalName>
    </None>
</ItemGroup>

<ItemGroup ...>
    <None Include="InfoCustomTwo.plist">
          <LogicalName>Info.plist</LogicalName>
    </None>
</ItemGroup>
```

The `iOS` folder looks like this where I still have the `Info.plist` but also other .plist files for other build configuration.

![image](https://github.com/user-attachments/assets/4563390c-2007-4089-b25d-4840d1bf1684)

The problem is this doesnt seem to be working. For example when I modified the `Bundle version` in the InfoCustomOne.plist had the configuration set for it where this .plist file was being used and did an publish/archive of the build it did not honor the `Bundle version` I provided but will reads from the generic `Info.plist`

**Comment by prabhav-mehra** [comment, marked as answer]
https://medium.com/@filipe.torres112/multi-platform-configuration-multiple-info-plist-and-manifest-xml-files-in-net-maui-14ef61acbfbe

**Reply by LeoJHarris** [reply]
@s3743761  How did I not find that but thanks heaps!! 😄

**Comment by disti80** [comment]
Talking about iOS (Info.plist) it seems to me that there are a some errors in the solution provided by the linked article.

1) Default Info.plist

In the article I read:

> It’s important to keep the original Info.plist file in your project for common configurations. The above code snippets remove and include the appropriate Info.plist files based on your build configuration, allowing you to manage distinct settings for Debug and Release builds while preserving the original Info.plist.

This seems to imply that I can have some settings in Info.plist that will be applied to _both_ Debug _and_ Release configurations, while settings in .Debug and .Release files will be applied to their specific configuration only.
But in .csproj file gets removed (``<None Remove="Platforms\iOS\Info.plist" />``) so common settings get lost.

2) wrong file names.

For debug configuration it should be:
```
<None Remove="Platforms\iOS\Info.plist" />
<None Include="Platforms\iOS\Info.Debug.plist"> <!-- not Info.plist -->
```
and for release:
```
<None Remove="Platforms\iOS\Info.plist" />
<None Include="Platforms\iOS\Info.Release.plist"> <!-- not Info.Debug.plist -->
```

I hope someone found a way to have common settings in "Info.plist" and specific settings in separate configuration-related files.
